client/src/App.tsx

```diff
@@ -66,30 +58,44 @@ const App: React.FC = () => {
   const [shortened, setShortened] = React.useState<number | null>(null)
   const [visits, setVisits] = React.useState<number | null>(null)
   const [shortenedCreated, setShortenedCreated] = React.useState('')
+  const [links, setLinks] = React.useState<string[]>([])
 
   const showToast = useCopyClipboardToast()
 
   const setShortenedCreatedArray = React.useMemo(() => [shortenedCreated], [shortenedCreated])
 
   // useMemo runs before useEffect (like the class constructor)
   React.useMemo(async () => {
-    const stats = await fetchStats()
+    const links: string[] = getLinksFromLocalStorage()
+    if (links.length) setLinks(links)
 
+    const stats = await fetchStats()
     if (stats) {
       setShortened(stats.totalShortened)
       setVisits(stats.totalVisits)
     }
   }, [])
 
+  const onLinkDelete = React.useCallback((url: string, links: string[]) => {
+    const linksFiltered = links.filter(link => link !== url)
+    localStorage.set(linksLocalStorageKey, [...linksFiltered])
```

Comment:
[Q] Why is it needed to make a copy of the array to pass to `localStorage.set`?

Reply:
😅 my bad I was doing the filter on the set function, and the moved to a constant (linksFiltered) and forgot to delete the spread